### PR TITLE
Prepare release v2.34.0

### PR DESCRIPTION
**Pull request by sdjmchattie**
Preparing release v2.34.0 in `master` for a production release of the recently reviewed work.

### CODE REVIEW

**Comment by sdjmchattie** [comment]
> What's the difference between this and #791? Would you add some notes in the PR description please?

Not very much.  This is preparing the `master` branch so that a production release is possible.

**Review by harrietc52 — APPROVED** [review]

File: lighthouse/helpers/plates.py [hunk 1]
```diff
@@ -215,8 +221,68 @@ def row_to_dict(row):
...
+    LOGGER.debug(f"Searching Sequencescape for plate with UUID: '{uuid}'.")
+
+    return request_with_retries(
+        request_func=lambda: requests.get(ss_url, params=params, headers=headers),
+        response_func=lambda response: len(response.json()["data"]) > 0,
+    )
+
+
+def filter_for_new_samples(samples: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    def is_sample_new(sample: Dict[str, str]) -> bool:
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply lighthouse/helpers/plates.py:264]: Fair enough.  We do have a lot of other instances of doing it this way and it does help the reader to identify the scope of the usage for this function, but I also understand why you might not like it.  I had always considered this approach as the method being part of the one it exists inside, hence it would be tested by the behaviour of the outer method.

File: tests/routes/v1/test_plates.py [hunk 2]
```diff
@@ -17,58 +17,197 @@
 NEW_PLATE_ENDPOINTS = [prefix + NEW_PLATE_ENDPOINT for prefix in ENDPOINT_PREFIXES]
 GET_PLATES_ENDPOINTS = [prefix + GET_PLATES_ENDPOINT for prefix in ENDPOINT_PREFIXES]
 
-FILTERED_PLATE_TYPES: List[Optional[str]] = [None, SS_PLATE_TYPE_DEFAULT]
-UNFILTERED_PLATE_TYPES: List[Optional[str]] = ["unfiltered"]
+FIT_TO_PICK_PLATE_TYPES: List[Optional[str]] = [None, SS_PLATE_TYPE_DEFAULT, "fit_to_pick_new_samples_only"]
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply tests/routes/v1/test_plates.py:20]: The checks during CI always ensure that the linting has been done project wide.  There is a configuration for the linting in this project to limit line length to 120 characters.  It might seem a lot, but we don't work on 80 character monitors these days, so I am inclined to agree with the configuration.  120 characters is not hard to parse on a modern display.
